QEfficient/cloud/infer.py

```diff
@@ -45,102 +44,166 @@ def main(
     ctx_len: int = 128,
     mxfp6: bool = False,
     mxint8: bool = False,
-    device_group: List[int] = [
-        0,
-    ],
+    device_group: List[int] = [0],
 ) -> None:
-    qpc_base_dir_name = (
-        f"qpc_{num_cores}cores_{batch_size}BS_{prompt_len}PL_{ctx_len}CL_{mos}MOS_"
-        + f"{len(device_group)}"
-        + "devices"
-        + ("_mxfp6_mxint8" if (mxfp6 and mxint8) else "_mxfp6" if mxfp6 else "_fp16_mxint8" if mxint8 else "_fp16")
+    """
+    Main function to download, optimize, compile, and execute a model on Cloud AI 100.
+
+    Parameters:
+    - model_name: The name or ID of the Hugging Face model.
+    - num_cores: Number of cores to compile on Cloud AI 100.
+    - prompt: Input prompt for text generation.
+    - prompts_txt_file_path: File path for input prompts from a text file.
+    - aic_enable_depth_first: Enable depth-first search during compilation.
+    - mos: Effort level to reduce on-chip memory.
+    - cache_dir: Directory to store Hugging Face downloads.
+    - hf_token: Hugging Face token for accessing private models.
+    - batch_size: Batch size for text generation.
+    - prompt_len: Sequence length for text generation.
+    - ctx_len: Context length for text generation.
+    - mxfp6: Compress constant MatMul weights to MXFP6 E2M3.
+    - mxint8: Compress Present/Past KV to MXINT8 using CustomIO config.
+    - device_group: Cloud AI 100 device IDs (comma-separated).
+    """
+
+    qpc_base_dir_name = create_qpc_base_dir_name(
+        num_cores, batch_size, prompt_len, ctx_len, mos, len(device_group), mxfp6, mxint8
     )
 
-    prompt = check_batch_size_and_num_prompts(prompt, prompts_txt_file_path, batch_size)
+    prompt = process_prompt(prompt, prompts_txt_file_path, batch_size)
 
-    # Get tokenizer
-    if hf_token is not None:
+    # Login to Hugging Face Hub if hf_token is provided
+    if hf_token:
         login(hf_token)
-    model_hf_path = hf_download(
-        repo_id=model_name,
-        cache_dir=cache_dir,
-        ignore_patterns=["*.txt", "*.onnx", "*.ot", "*.md", "*.tflite", "*.pdf", "*.msgpack", "*.h5"],
-    )
-    tokenizer = AutoTokenizer.from_pretrained(
-        model_hf_path, use_cache=True, padding_side="left", trust_remote_code=True
-    )
 
+    # Check if pre-compiled QPC/Onnx exists and execute if so
     qpc_path_exists, qpc_dir_path = qpc_exists(model_name, qpc_base_dir_name)
+    onnx_path_exists, onnx_dir_path, onnx_model_path = onnx_exists(model_name)
+
     if qpc_path_exists:
```

Comment:
Why this condition to decide if we need to download the tokenizer files?
We always need tokenizer irrespective of the path of the code execution, i.e. for exec_kv function. Which will always be called.

Reply:
Yes, just need tokenizer for the execution, so if user wants to delete cache and run the infer for execution, this will download tokenizer only and go ahead with execution.

Reply:
yes, agree, `load_tokenizer`() should always be on.